Annotate CSS/Style numeric primitives with their value range

weinig opened the pull request:
#### 65b8b27448c2291db57953908cc3814051f188ab
<pre>
Annotate CSS/Style numeric primitives with their value range
<a href="https://bugs.webkit.org/show_bug.cgi?id=281080">https://bugs.webkit.org/show_bug.cgi?id=281080</a>

Reviewed by Darin Adler.

Adds a non-type template parameter to each CSS/Style numeric primitive
type, encoding the allowed value range directly. The value range is
represented by CSS::Range. Eventually it will likely be useful to extend
this to additional metadata, such as whether unitless values are supported
when parsing, but this initial change only introduces the range.

Most users of the numeric types will use them in non-generic ways (e.g.
the gradient code uses a CSS::LengthPercentage&lt;CSS::NonNegative&gt;), but
for the support code to parse and resolve, some functionality was pulled
into headers as required for the templates. This lead to a good opportunity
to unify and simplify the parsing code, with shared implementations of
the token specific consumers now being used by all the numeric types.

A few additional related changes made:
- Remove now unused ValueRange parameter from the parser options.
- Remove already unused NegativePercentagePolicy from the parser options.
- Used CSS::Range throughout calc code to ensure proper clamping.
- Removed support for unitless &lt;time&gt; values, which were only being
  used internally by the marquee element code (but was easily fixable).

Adopted spec&apos;d range values in position, gradient and font code. Of
particular note is the font code, where we were able to remove quite
a bit of custom validation code for complex ranges types like:

    &lt;number [1,1000]&gt;

and

    &lt;angle [-90deg,90deg]&gt;

* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSFilterFunctionDescriptor.h:
* Source/WebCore/css/CSSFontFaceSet.cpp:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/calc/CSSCalcTree+CalculationValue.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Parser.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Parser.h:
* Source/WebCore/css/calc/CSSCalcTree+Serialization.cpp:
* Source/WebCore/css/calc/CSSCalcTree.h:
* Source/WebCore/css/calc/CSSCalcValue.cpp:
* Source/WebCore/css/calc/CSSCalcValue.h:
* Source/WebCore/css/color/CSSColorConversion+Normalize.h:
* Source/WebCore/css/color/CSSColorConversion+ToTypedColor.h:
* Source/WebCore/css/color/CSSColorDescriptors.h:
* Source/WebCore/css/color/CSSColorMixResolver.cpp:
* Source/WebCore/css/color/CSSColorMixResolver.h:
* Source/WebCore/css/color/CSSColorMixSerialization.cpp:
* Source/WebCore/css/color/CSSColorMixSerialization.h:
* Source/WebCore/css/color/CSSUnresolvedColor.h:
* Source/WebCore/css/color/CSSUnresolvedColorMix.cpp:
* Source/WebCore/css/color/CSSUnresolvedColorMix.h:
* Source/WebCore/css/color/StyleColorMix.h:
* Source/WebCore/css/parser/CSSParserFastPaths.cpp:
* Source/WebCore/css/parser/CSSParserFastPaths.h:
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Angle.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Angle.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+AngleDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+AnglePercentage.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+AnglePercentage.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+AnglePercentageDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+CSSPrimitiveValueResolver.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Color.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+FlexDefinitions.h: Copied from Source/WebCore/css/parser/CSSPropertyParserConsumer+Integer.h.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+FrequencyDefinitions.h: Copied from Source/WebCore/style/values/StylePosition.h.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Image.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Integer.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Integer.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+IntegerDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Length.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Length.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+LengthDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+LengthPercentage.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+LengthPercentage.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+LengthPercentageDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+MetaConsumer.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+MetaConsumerDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Number.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Number.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+NumberDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Percentage.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+PercentageDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Position.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Position.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Primitives.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Primitives.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Resolution.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Resolution.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+ResolutionDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Time.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Time.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+TimeDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
* Source/WebCore/css/parser/CSSPropertyParserOptions.h:
* Source/WebCore/css/process-css-properties.py:
* Source/WebCore/css/typedom/CSSNumericValue.cpp:
* Source/WebCore/css/typedom/CSSUnitValue.cpp:
* Source/WebCore/css/typedom/numeric/CSSMathValue.cpp:
* Source/WebCore/css/values/CSSGradient.cpp:
* Source/WebCore/css/values/CSSGradient.h:
* Source/WebCore/css/values/CSSPosition.cpp:
* Source/WebCore/css/values/CSSPosition.h:
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+ComputedStyleDependencies.cpp:
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+ComputedStyleDependencies.h:
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+SymbolReplacement.cpp:
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+SymbolReplacement.h:
* Source/WebCore/css/values/CSSPrimitiveNumericTypes.cpp: Added.
* Source/WebCore/css/values/CSSPrimitiveNumericTypes.h:
* Source/WebCore/html/HTMLInputElement.cpp:
* Source/WebCore/html/HTMLMarqueeElement.cpp:
* Source/WebCore/platform/Length.cpp:
* Source/WebCore/platform/calc/CalculationTree.h:
* Source/WebCore/platform/calc/CalculationValue.cpp:
* Source/WebCore/platform/calc/CalculationValue.h:
* Source/WebCore/rendering/style/StyleGradientImage.cpp:
* Source/WebCore/style/StyleResolveForFont.cpp:
* Source/WebCore/style/values/StyleGradient.cpp:
* Source/WebCore/style/values/StyleGradient.h:
* Source/WebCore/style/values/StylePosition.cpp:
* Source/WebCore/style/values/StylePosition.h:
* Source/WebCore/style/values/StylePrimitiveNumericTypes+Conversions.cpp:
* Source/WebCore/style/values/StylePrimitiveNumericTypes+Conversions.h:
* Source/WebCore/style/values/StylePrimitiveNumericTypes.cpp: Removed.
* Source/WebCore/style/values/StylePrimitiveNumericTypes.h:

Canonical link: <a href="https://commits.webkit.org/285128@main">https://commits.webkit.org/285128@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e3ba0970b0bc3988e290ceec0667d367fa9bbdbc

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/71656 "61 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/51069 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/24430 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/75770 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/22861 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/73771 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/58870 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/22681 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/56597 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/15077 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/74722 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/46334 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/61725 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/37046 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/42995 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/19192 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/21202 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/64899 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/19555 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/77490 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/15890 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/18737 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/64309 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/15933 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/61758 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/64309 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15824 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/12459 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/6097 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/46869 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/1648 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/47940 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/49224 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/47682 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/css/calc/CSSCalcTree+Serialization.cpp:
```diff
@@ -245,10 +245,10 @@ static Vector<ChildRepresentation, 16> generateSortedChildrenMap(const Children&
 {
     value = std::isnan(value) ? 0 : value;
-    return range == ValueRange::NonNegative && value < 0 ? 0 : value;
+    return std::clamp(value, range.min, range.max);
```
I normally avoid `std::clamp` because it returns `max` if `max` < `min`, and we typically want `min` in those cases in most call sites. But if that is not an issue here because `CSS::Range` never violates the invariant, then I think it's great to use it.

darinadler commented on Source/WebCore/css/calc/CSSCalcValue.cpp:
```diff
@@ -188,7 +189,7 @@ inline double CSSCalcValue::clampToPermittedRange(double value) const
     if (m_tree.category == Calculation::Category::Integer)
         value = std::floor(value + 0.5);
 
-    return m_tree.range == ValueRange::NonNegative && value < 0 ? 0 : value;
```
I normally avoid `std::clamp` because it returns `max` if `max` < `min`, and we typically want `min` in those cases in most call sites. But if that is not an issue here because `CSS::Range` never violates the invariant, then I think it's great to use it.

darinadler commented on Source/WebCore/css/color/StyleColorMix.h:
```diff
@@ -39,7 +39,7 @@ struct StyleColorMix {
 
     struct Component {
-        using Percentage = Style::Percentage;
+        using Percentage = Style::Percentage<CSS::Range{0, 100}>;
```
This idiom comes up often enough that it seems we might want a named type.

darinadler commented on Source/WebCore/css/parser/CSSPropertyParserConsumer+AngleDefinitions.h:
```diff
@@ -24,42 +24,39 @@
-    static constexpr CSSParserTokenType tokenType = NumberToken;
-    static std::optional<CSS::AngleRaw> consume(CSSParserTokenRange&, const CSSParserContext&, CSSCalcSymbolsAllowed, CSSPropertyParserOptions);
+struct AngleValidator {
+    static bool isValid(CSSUnitType unitType, CSSPropertyParserOptions)
```
I'd be so tempted to write `constexpr` here.

darinadler commented on Source/WebCore/css/parser/CSSPropertyParserConsumer+AnglePercentage.h:
```diff
@@ -25,21 +25,20 @@
 #pragma once
 
 #include "CSSPropertyParserOptions.h"
+#include "Length.h"
```
Why is this being added?

weinig replied:
Removed it from CSSPropertyParserOptions.h, needed for the default argument value in a function below.

darinadler commented on Source/WebCore/css/parser/CSSPropertyParserConsumer+FlexDefinitions.h:
```diff
@@ -0,0 +1,58 @@
+    using FunctionToken = FunctionConsumerForCalcValues<CSS::Flex<R>>;
+    using DimensionToken = DimensionConsumer<CSS::Flex<R>, FlexValidator>;
+
+    // NOTE: Flex does not allow unitless consumers.
```
Does this comment mean it should not, but currently does?

weinig replied:
It is supposed to mean that it both does not and should never (only angle and length should allow unitless). I’ll make it a bit more clear.

darinadler commented on Source/WebCore/css/parser/CSSPropertyParserConsumer+Length.h:
```diff
@@ -25,20 +25,19 @@
 #pragma once
 
 #include "CSSPropertyParserOptions.h"
+#include "Length.h"
```
I know there must be a good reason, I just can't see what it is.

darinadler commented on Source/WebCore/css/values/CSSPrimitiveNumericTypes+ComputedStyleDependencies.h:
```diff
@@ -37,11 +37,27 @@ namespace CSS {
-template<> struct ComputedStyleDependenciesCollector<LengthRaw> { void operator()(ComputedStyleDependencies&, const LengthRaw&); };
-template<> struct ComputedStyleDependenciesCollector<LengthPercentageRaw> { void operator()(ComputedStyleDependencies&, const LengthPercentageRaw&); };
+template<auto R> struct ComputedStyleDependenciesCollector<LengthRaw<R>> {
+    inline void operator()(ComputedStyleDependencies& dependencies, const LengthRaw<R>& value)
```
Aren't members with bodies inline by default? Does the keyword do anything here?

weinig replied:
Yeah, you are right. Hold over from when they were out of line (which I didn’t end up with as the syntax was way to noisy and ended up less readable. )

will fix.

darinadler commented on Source/WebCore/css/values/CSSPrimitiveNumericTypes+ComputedStyleDependencies.h:
```diff
@@ -37,11 +37,27 @@ namespace CSS {
+    }
+};
+template<auto R> struct ComputedStyleDependenciesCollector<LengthPercentageRaw<R>> {
+    inline void operator()(ComputedStyleDependencies& dependencies, const LengthPercentageRaw<R>& value)
```
Aren't members with bodies inline by default? Does the keyword do anything here?

darinadler commented on Source/WebCore/css/values/CSSPrimitiveNumericTypes.h:
```diff
@@ -33,6 +33,24 @@
+inline constexpr auto All = Range { -Range::infinity, Range::infinity };
+
+// Constant value for `[0,∞]`.
+inline constexpr auto NonNegative = Range { 0, Range::infinity };
```
Since nonnegative is a word, we don't need to capitalize the second N.

weinig committed:
Annotate CSS/Style numeric primitives with their value range
https://bugs.webkit.org/show_bug.cgi?id=281080

Reviewed by Darin Adler.

Adds a non-type template parameter to each CSS/Style numeric primitive
type, encoding the allowed value range directly. The value range is
represented by CSS::Range. Eventually it will likely be useful to extend
this to additional metadata, such as whether unitless values are supported
when parsing, but this initial change only introduces the range.

Most users of the numeric types will use them in non-generic ways (e.g.
the gradient code uses a CSS::LengthPercentage<CSS::NonNegative>), but
for the support code to parse and resolve, some functionality was pulled
into headers as required for the templates. This lead to a good opportunity
to unify and simplify the parsing code, with shared implementations of
the token specific consumers now being used by all the numeric types.

A few additional related changes made:
- Remove now unused ValueRange parameter from the parser options.
- Remove already unused NegativePercentagePolicy from the parser options.
- Used CSS::Range throughout calc code to ensure proper clamping.
- Removed support for unitless <time> values, which were only being
  used internally by the marquee element code (but was easily fixable).

Adopted spec'd range values in position, gradient and font code. Of
particular note is the font code, where we were able to remove quite
a bit of custom validation code for complex ranges types like:

    <number [1,1000]>

and

    <angle [-90deg,90deg]>

* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSFilterFunctionDescriptor.h:
* Source/WebCore/css/CSSFontFaceSet.cpp:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/calc/CSSCalcTree+CalculationValue.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Parser.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Parser.h:
* Source/WebCore/css/calc/CSSCalcTree+Serialization.cpp:
* Source/WebCore/css/calc/CSSCalcTree.h:
* Source/WebCore/css/calc/CSSCalcValue.cpp:
* Source/WebCore/css/calc/CSSCalcValue.h:
* Source/WebCore/css/color/CSSColorConversion+Normalize.h:
* Source/WebCore/css/color/CSSColorConversion+ToTypedColor.h:
* Source/WebCore/css/color/CSSColorDescriptors.h:
* Source/WebCore/css/color/CSSColorMixResolver.cpp:
* Source/WebCore/css/color/CSSColorMixResolver.h:
* Source/WebCore/css/color/CSSColorMixSerialization.cpp:
* Source/WebCore/css/color/CSSColorMixSerialization.h:
* Source/WebCore/css/color/CSSUnresolvedColor.h:
* Source/WebCore/css/color/CSSUnresolvedColorMix.cpp:
* Source/WebCore/css/color/CSSUnresolvedColorMix.h:
* Source/WebCore/css/color/StyleColorMix.h:
* Source/WebCore/css/parser/CSSParserFastPaths.cpp:
* Source/WebCore/css/parser/CSSParserFastPaths.h:
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Angle.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Angle.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+AngleDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+AnglePercentage.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+AnglePercentage.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+AnglePercentageDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+CSSPrimitiveValueResolver.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Color.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+FlexDefinitions.h: Copied from Source/WebCore/css/parser/CSSPropertyParserConsumer+Integer.h.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+FrequencyDefinitions.h: Copied from Source/WebCore/style/values/StylePosition.h.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Image.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Integer.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Integer.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+IntegerDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Length.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Length.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+LengthDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+LengthPercentage.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+LengthPercentage.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+LengthPercentageDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+MetaConsumer.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+MetaConsumerDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Number.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Number.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+NumberDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Percentage.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+PercentageDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Position.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Position.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Primitives.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Primitives.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Resolution.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Resolution.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+ResolutionDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Time.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Time.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+TimeDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
* Source/WebCore/css/parser/CSSPropertyParserOptions.h:
* Source/WebCore/css/process-css-properties.py:
* Source/WebCore/css/typedom/CSSNumericValue.cpp:
* Source/WebCore/css/typedom/CSSUnitValue.cpp:
* Source/WebCore/css/typedom/numeric/CSSMathValue.cpp:
* Source/WebCore/css/values/CSSGradient.cpp:
* Source/WebCore/css/values/CSSGradient.h:
* Source/WebCore/css/values/CSSPosition.cpp:
* Source/WebCore/css/values/CSSPosition.h:
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+ComputedStyleDependencies.cpp:
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+ComputedStyleDependencies.h:
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+SymbolReplacement.cpp:
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+SymbolReplacement.h:
* Source/WebCore/css/values/CSSPrimitiveNumericTypes.cpp: Added.
* Source/WebCore/css/values/CSSPrimitiveNumericTypes.h:
* Source/WebCore/html/HTMLInputElement.cpp:
* Source/WebCore/html/HTMLMarqueeElement.cpp:
* Source/WebCore/platform/Length.cpp:
* Source/WebCore/platform/calc/CalculationTree.h:
* Source/WebCore/platform/calc/CalculationValue.cpp:
* Source/WebCore/platform/calc/CalculationValue.h:
* Source/WebCore/rendering/style/StyleGradientImage.cpp:
* Source/WebCore/style/StyleResolveForFont.cpp:
* Source/WebCore/style/values/StyleGradient.cpp:
* Source/WebCore/style/values/StyleGradient.h:
* Source/WebCore/style/values/StylePosition.cpp:
* Source/WebCore/style/values/StylePosition.h:
* Source/WebCore/style/values/StylePrimitiveNumericTypes+Conversions.cpp:
* Source/WebCore/style/values/StylePrimitiveNumericTypes+Conversions.h:
* Source/WebCore/style/values/StylePrimitiveNumericTypes.cpp: Removed.
* Source/WebCore/style/values/StylePrimitiveNumericTypes.h:

Canonical link: https://commits.webkit.org/285128@main
